eva-server/src/main/java/uk/ac/ebi/eva/server/ws/ga4gh/GA4GHVariantWSServer.java

```diff
@@ -73,35 +83,40 @@ public GASearchVariantsResponse getVariantsByRegion(@RequestParam("referenceName
                                         @RequestParam(name = "pageSize", defaultValue = "10") int limit)
             throws IllegalOpenCGACredentialsException, UnknownHostException, IOException {
         initializeQueryOptions();
-        
-        VariantDBAdaptor variantMongoDbAdaptor = DBAdaptorConnector.getVariantDBAdaptor("hsapiens_grch37");
-        
+
+        MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName("hsapiens_grch37"));
+
         if (files != null && !files.isEmpty()) {
             queryOptions.put("files", files);
         }
-        
+        List<VariantEntityRepositoryFilter> filters = Helpers.getVariantEntityRepositoryFilters(null, null, null, null,
+                                                                                                null, files, null, null);
+
         int idxCurrentPage = 0;
         if (pageToken != null && !pageToken.isEmpty() && StringUtils.isNumeric(pageToken)) {
             idxCurrentPage = Integer.parseInt(pageToken);
             queryOptions.put("skip", idxCurrentPage * limit);
         }
         queryOptions.put("limit", limit);
-        
-        // Create the provided region, whose size can't exceed 1 million positions
+
         Region region = new Region(chromosome, start, end);
-        int regionSize = region.getEnd()-region.getStart();
-        
-        if (regionSize > 1000000) {
```

Comment:
Yeah, due to the new indexing and the method the VariantEntityRepositoryImpl queries the database.
@cyenyxe can confirm if I'm correct in thinking this.